refactor(docs): refactor front-matter to use mapping instead of switch cases

RishhiB opened the pull request:
just a small refactor to use mapping

justus-camp committed:
Use IdCompressor for RevisionTags (#18775)

## Description
This PR swaps out UUIDs for IDs generated by the runtime IdCompressor.

---------

Co-authored-by: Justus Camp <justuscamp@Justuss-MacBook-Pro.local>
Co-authored-by: Abram Sanderson <absander@microsoft.com>
Co-authored-by: Yann Achard <achardy@microsoft.com>

DLehenbauer committed:
api: manually trim 'fluid-framework' via exports (#18985)

Trims the types re-exported by 'fluid-framework' by hand to only include
only those that are tagged public:

For packages other than tree, I did this by adding the '/public' suffix
to the import and removing the types that no longer resolved.

For tree, I scraped the list of types from the generated public.d.ts.

'shared-tree-demo' has been updated to import from 'fluid-framework' for validation that the produced *.d.ts is consumable by tsc.

Abe27342 committed:
Decrease pipeline timeout back to 60 minutes (#18993)

## Description

Reverts the timeout increase implemented in #18831 and add some notes
for posterity on when we might need to bump it again.

The bulk of the reason for this increase was increased numbers of e2e
full compat tests running. This is dependent on number of previous
major/minor/etc versions that we test against, which was reduced with
the upcoming release candidate changes (we don't test the current build
against all of 2.0.0-internal.x.0.0, just some of it).

Co-authored-by: Abram Sanderson <absander@microsoft.com>

CraigMacomber committed:
tree: Audit editManagerFormat, add version (#18992)

## Description

Move some non format stuff out of EditManagerFormat, and add a version
number and use makeVersionedValidatedCodec.

Document some issues found while working on this as TODOs.

## Breaking Changes

Breaks document format compatibility.

zhenmichael committed:
enable build multiversion docs (#18996)

enable build multiversion for build docs

markfields committed:
GC: Move detection of added outbound routes to the ContainerRuntime instead of the DDS layer (#18456)

This moves the detection of added outbound routes to the
ContainerRuntime instead of the DDS layer. Two main advantages:

1. Detection happens immediately upon op receipt, instead of having to
wait until the DataStore/DDS loads
2. We don't get false notifications every time the serialized DDS is parsed (which happens on load), only
when it's first received.

daesunp committed:
add end to end chunk encoding test (#18912)

## Description

This PR adds an end to end chunk encoding test for sharedTree. The tests
are skipped as there is currently no way to insert a uniform chunk into
a tree.

---------

Co-authored-by: Craig Macomber (Microsoft) <42876482+CraigMacomber@users.noreply.github.com>

tylerbutler committed:
build: Use .cjs file extension with CJS eslint configs (#18990)

Renames all the .eslintrc.js files to use the .cjs file extension. This
will make it easier to add type: module to ESM-only packages.

jason-ha committed:
improvement(tree): explicit internal paths (#18989)

explicit paths to support moduleResolution node16+ with tsc

import/no-internal-modules fully disabled (lots of internal module paths
already that weren't caught)

kekachmar committed:
remove user details from quorum members (#19000)

Sets the user property of the IScribeProtocolState to null to remove
this information from all checkpoints
This is so we do not store any personably identifiable information
stored with the metadata in our checkpoints

yann-achard-MS committed:
test(tree): repro for fuzz bug (#19004)

yann-achard-MS committed:
test(tree): less constrained fuzz move (#19006)

yann-achard-MS committed:
test(tree): add repro for dependency tracking bug (#19008)

yann-achard-MS committed:
refactor(tree): use field cursors in low-level editing API (#19009)

yann-achard-MS committed:
refactor(tree): Change delete terminology to remove (#19011)

justus-camp committed:
Tests for some move scenarios (#18703)

## Description

Adds a couple tests described by AB#6375. Let me know if there are more
scenarios here we want to test.

## Reviewer Guidance

The review process is outlined on [this wiki
page](https://github.com/microsoft/FluidFramework/wiki/PR-Guidelines#guidelines).

agarwal-navin committed:
Added tests to validate that incremental summary works correctly (#19007)

Added couple of tests for scenarios where a data store / DDS should be
summarized as a handle instead of tree. It also validates that the
handle id is correct.

Also, added the handle id validation to incremental summary context
tests which test incremental summary at DDS level.

agarwal-navin committed:
Remove initialSummary and throwOnFailure from summarizer nodes (#19001)

`initialSummary` and `throwOnFailure` are no longer used by summarizer
node. Removing them and the logic around them.

AB#3738

CraigMacomber committed:
tree: Make use of wrongSchema more local and explicit (#19019)

## Description

This makes it more clear which tests need to have valid schema added.

taylorsw04 committed:
id-compressor: Unit test ghost session API (#19021)

## Description

This PR adds unit tests for the ghost session API in the ID compressor.

CraigMacomber committed:
tree: Cleanup tests using WrongSchema (#19024)

## Description

Simplify tests which are using WrongSchema to use simpler schema, better
validation libraries, and actually have schema.

yann-achard-MS committed:
refactor(tree): consolidate related tests (#19012)

jason-ha committed:
tool(tree): enforce internal imports via index.js (#19023)

fix the disables for schemaEvolutionExamples.spec.ts and
schemaAware.spec.ts

agarwal-navin committed:
Code cleanup in container runtime around refresh from summary logic (#19002)

Cleaned up the following from container runtime. These were leftovers
from the logic to update the latest summary by downloading a snapshot.
The PR that changed the logic -
https://github.com/microsoft/FluidFramework/pull/17185.
Before the above PR, when `refreshLatestAck` summary option was true,
the latest snapshot was downloaded and the latest summary state was
updated from that. After the above PR, we never update the latest
summary state from a snapshot, instead we close the summarizer in this
scenario.

- `latestSnapshotVersionId` is no longer used in `submitSummary` and has
been removed.
- In `refreshLatestSummaryAck`, if a snapshot is fetched and is not the
latest, a data processing error is thrown. However, this is no longer
needed because the summarizer is closed in this scenario anyway. The
snapshot fetch is just an optimization and throwing an error is
redundant.

tylerbutler committed:
feat(fluid-build): Task caching for ts2esm tasks (#19027)

This change adds task handling for ts2esm tasks to fluid-build. This
will enable incremental build with ts2esm tasks. Also updates the
fluid-build task config for our repo to understand the new task, but it
won't have any effect until tasks are added to individual package.jsons
in a later change.

---------

Co-authored-by: Jason Hartman <jasonha@microsoft.com>

jason-ha committed:
tool(tree): fix internal imports via index.js (#19028)

or suppress test import rule

CraigMacomber committed:
tree2: Remove old fuzz test repro (#19025)

## Description

The removed test was skipped, and was used to debug errors in a specific
fuzz test. The test is now passing, and is a bit overly complex to be
worth keeping as a regression test but should be covered by the now
passing fuzz tests.

Josmithr committed:
fix(docs): Render config (#19033)

The config used for rendering API docs included 2 obsolete parameters:
`fileNamePolicy` (which was previously renamed to `getFileNameForItem`)
and `filterPolicy` (which was previously renamed to `skipPackage`).

This PR...
1. fixes the former by removing rule override altogether (we were
specifying the default policy anyways).
2. fixes the latter property name and adds `@fluid-private` to the
package namespaces it skips, since we don't want these docs to be
public-facing.

RishhiB committed:
refactor(docs): refactor front-matter to use mapping instead of switch cases

RishhiB committed:
format

RishhiB committed:
use direct method call instead of intermediate variable

RishhiB committed:
small renaming

tylerbutler committed:
build(client): Bump build-tools to latest prerelease version (#19040)

Updated the following:

  client (release group)

Dependencies on build-tools updated:

  @fluid-tools/build-cli: 0.29.0-227016
  @fluidframework/build-tools: 0.29.0-227016
  @fluidframework/bundle-size-tools: 0.29.0-227016
  @fluid-tools/version-tools: 0.29.0-227016

Command used:

```shell
flub bump deps build-tools --updateChecker homegrown -g client --prerelease
```

jason-ha committed:
build(client-tree): ts2esm support and enforcement (#18998)

ts2esm is run as part of build. It may change files which will cause CI
to fail.
Changed files may need to be reformatted, which is not done
automatically.

daesunp committed:
unskip end to end chunk insert op test (#19026)

## Description

This PR unskips the end to end chunk insert op test, as it is now
working due to [PR
19009](https://github.com/microsoft/FluidFramework/pull/19009).

CraigMacomber committed:
tree: Remove simple-tree insertable (#19032)

## Description

Finish rename of TreeNodeSchema to FlexTreeNodeSchema to avoid colliding
with other TreeNodeSchema.
Remove schema aware insertable types which use FlexTreeNodeSchema to
make types for the simple-tree layer (which uses its own schema system
which has strong insertable types in class-tree)

Josmithr committed:
feat(docs): Add version string to front-matter in generated API docs (#19035)

This is in preparation for publishing multiple versions of API docs on
the website. Future changes will utilize this metadata to differentiate
between API documents at different versions.

Also adds some function documentation.

Josmithr commented on docs/api-markdown-documenter/front-matter.js:
```diff
@@ -20,95 +20,77 @@ const generatedContentNotice =
+		version,
+		kind: apiItem.kind,
+		members: new Map(),
+		package: associatedPackage ?
+			associatedPackage.name.replace(/"/g, "").replace(/!/g, "") :
+			"undefined",
```
```suggestion
			undefined,
```

Josmithr replied:
```suggestion
		package: associatedPackage?.name.replace(/"/g, "").replace(/!/g, "") 
```

Josmithr commented on docs/api-markdown-documenter/front-matter.js:
```diff
@@ -20,95 +20,77 @@ const generatedContentNotice =
+		[ApiItemKind.Package]: `${baseTitle} Package`,
+		[ApiItemKind.Namespace]: `${baseTitle} Namespace`,
+	};
+	return kindTitleMap[apiItem.kind] || baseTitle;
```
Nit
```suggestion
	return kindTitleMap[apiItem.kind] ?? baseTitle;
```

CraigMacomber committed:
tree: Combine class-tree into simple-tree (#19050)

## Description

Now that the last schema aware types using the flex-tree schema system
have been removed from simple-tree, combine it with class tree.

Also removes IntegerIndexable, which was unused as it seems unlikely
that a mostly proxyless refactor of simple-tree into more orthogonal
+testable components is not going to happen any time soon.

Josmithr commented on docs/api-markdown-documenter/front-matter.js:
```diff
@@ -20,95 +20,77 @@ const generatedContentNotice =
-			customRenderers,
+	apiMembers.filter(({ displayName }) => displayName !== "")
+		.forEach(api => {
+			frontMatter.members[api.kind] = frontMatter.members[api.kind] || {};
```
Nit
```suggestion
			frontMatter.members[api.kind] = frontMatter.members[api.kind] ?? {};
```

jason-ha committed:
tool(eslint-config-fluid): allow index.js import (#19046)

With explicit import paths `/index.js` should be generally allowed as
that was allowed by linter when `/` would resolve to `/index.js`.

Bump version for release.

Josmithr reviewed (approved):
Changes look good to me! Thanks for the cleanup!

You'll want to merge main to pick up my most recent changes here, but otherwise should be good to go! 🚀

RishhiB committed:
Update docs/api-markdown-documenter/front-matter.js

Co-authored-by: Joshua Smithrud <54606601+Josmithr@users.noreply.github.com>

RishhiB committed:
Update docs/api-markdown-documenter/front-matter.js

Co-authored-by: Joshua Smithrud <54606601+Josmithr@users.noreply.github.com>

taylorsw04 committed:
id-compressor: Verify stateful takeNextRange behavior (#19051)

[How contribute to this
repo](https://github.com/microsoft/FluidFramework/blob/main/CONTRIBUTING.md).

[Guidelines for Pull
Requests](https://github.com/microsoft/FluidFramework/wiki/PR-Guidelines#guidelines).

## Description

This PR confirms that the compressor maintained a stateful takeNextRange
through the recent rewrite (it was suspected that it had not).

RishhiB committed:
Update docs/api-markdown-documenter/front-matter.js

Co-authored-by: Joshua Smithrud <54606601+Josmithr@users.noreply.github.com>

taylorsw04 committed:
id-compressor: Remove deprecated compressor types from runtime definitions (#19031)

## Description

This removes re-exported/deprecated compressor types from the runtime
definitions public API. They were moved to the id-compressor package
earlier.

## Breaking Changes

The removed APIs were technically public but are very unlikely to ever
have been used. They were moved to a publicly available package in
#18749.

RishhiB commented:
oh no, got issues merging main in
